compiler/rustc_middle/src/query/mod.rs

```diff
@@ -953,12 +953,17 @@ rustc_queries! {
         desc { "get a &core::panic::Location referring to a span" }
     }
 
+    // FIXME get rid of this with valtrees
     query lit_to_const(
         key: LitToConstInput<'tcx>
     ) -> Result<ty::Const<'tcx>, LitToConstError> {
         desc { "converting literal to const" }
     }
 
+    query lit_to_constant(key: LitToConstInput<'tcx>) -> Result<mir::ConstantKind<'tcx>, LitToConstError> {
```

Comment:
still think we want to remove that query, this method is pretty cheap and only called in the same module afaict.

Using it as a method directly seems better to me